test/e2e/storage/drivers/in_tree.go

```diff
@@ -426,11 +437,83 @@ func (h *hostPathDriver) CreateVolume(ctx context.Context, config *storageframew
 	f := config.Framework
 	cs := f.ClientSet
 
+	targetPath := fmt.Sprintf("/tmp/%v", f.Namespace.Name)
+	volumeName := "test-volume"
+
 	// pods should be scheduled on the node
 	node, err := e2enode.GetRandomReadySchedulableNode(ctx, cs)
 	framework.ExpectNoError(err)
 	config.ClientNodeSelection = e2epod.NodeSelection{Name: node.Name}
-	return nil
+
+	cmd := fmt.Sprintf("mkdir %v -m 777", targetPath)
+	privileged := true
+
+	// Launch pod to initialize hostPath directory
+	prepPod := &v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: fmt.Sprintf("hostpath-prep-%s", f.Namespace.Name),
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name:    fmt.Sprintf("init-volume-%s", f.Namespace.Name),
+					Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+					Command: []string{"/bin/sh", "-ec", cmd},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      volumeName,
+							MountPath: "/tmp",
+						},
+					},
+					SecurityContext: &v1.SecurityContext{
+						Privileged: &privileged,
+					},
+				},
+			},
+			RestartPolicy: v1.RestartPolicyNever,
+			Volumes: []v1.Volume{
+				{
+					Name: volumeName,
+					VolumeSource: v1.VolumeSource{
+						HostPath: &v1.HostPathVolumeSource{
+							Path: "/tmp",
+						},
+					},
+				},
+			},
+			NodeName: node.Name,
+		},
+	}
+	// h.prepPod will be reused in cleanupDriver.
+	pod, err := f.ClientSet.CoreV1().Pods(f.Namespace.Name).Create(ctx, prepPod, metav1.CreateOptions{})
+	framework.ExpectNoError(err, "while creating hostPath init pod")
+
+	err = e2epod.WaitForPodSuccessInNamespaceTimeout(ctx, f.ClientSet, pod.Name, pod.Namespace, f.Timeouts.PodStart)
+	framework.ExpectNoError(err, "while waiting for hostPath init pod to succeed")
+
+	err = e2epod.DeletePodWithWait(ctx, f.ClientSet, pod)
+	framework.ExpectNoError(err, "while deleting hostPath init pod")
+	return &hostPathVolume{
+		targetPath: targetPath,
+		prepPod:    prepPod,
+		f:          f,
+	}
+}
+
+func (v *hostPathVolume) DeleteVolume(ctx context.Context) {
```

Comment:
Where is this being called?

Reply:
@pohly 
1. https://github.com/carlory/kubernetes/blob/master/test/e2e/storage/testsuites/subpath.go#L178
2. https://github.com/carlory/kubernetes/blob/master/test/e2e/storage/framework/volume_resource.go#L245